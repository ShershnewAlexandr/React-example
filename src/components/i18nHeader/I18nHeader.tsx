import React from 'react';
import { Grid } from '@material-ui/core';
import { supportedLanguages } from '../../i18n/i18n';
import './I18nHeaderSyles.css';

interface I18nHeaderProps {
  changeLang: (lang: string) => (e: React.MouseEvent) => void;
}

function I18nHeader({ changeLang }: I18nHeaderProps) {
  return (
    <div className="i18nHeader-container">
      <Grid container direction="row-reverse">
        {supportedLanguages.map(lang => (
          <Grid item key={lang.code}>
            <div
              data-lang={lang.code}
              className="i18nHeader-icon-container"
              onClick={changeLang(lang.code)}
            >
              <img className="i18nHeader-icon" src={lang.icon} alt="lang" />
            </div>
          </Grid>
        ))}
      </Grid>
    </div>
  );
}

export default I18nHeader;
