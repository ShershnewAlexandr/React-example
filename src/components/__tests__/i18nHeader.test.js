import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import I18nHeader from '../i18nHeader/I18nHeader';

describe('i18nHeader', () => {
  const props = {
    changeLang: jest.fn(),
  };

  const i18nHeader = shallow(<I18nHeader {...props} />);

  it('snapshot', () => {
    expect(toJson(i18nHeader)).toMatchSnapshot();
  });

  it('should init 3 lang', () => {
    expect(props.changeLang).toBeCalledTimes(3);
  });
});
