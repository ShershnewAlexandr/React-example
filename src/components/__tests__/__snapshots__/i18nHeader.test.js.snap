// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18nHeader snapshot 1`] = `
<div
  className="i18nHeader-container"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="row-reverse"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="en"
    >
      <div
        className="i18nHeader-icon-container"
        data-lang="en"
      >
        <img
          alt="lang"
          className="i18nHeader-icon"
          src="united_kingdom_round_icon_64.png"
        />
      </div>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="ru"
    >
      <div
        className="i18nHeader-icon-container"
        data-lang="ru"
      >
        <img
          alt="lang"
          className="i18nHeader-icon"
          src="russia_round_icon_64.png"
        />
      </div>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="fr"
    >
      <div
        className="i18nHeader-icon-container"
        data-lang="fr"
      >
        <img
          alt="lang"
          className="i18nHeader-icon"
          src="france_round_icon_64.png"
        />
      </div>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
